Add type_of_union to check_null_field when decoding union types.

chaidao001 opened the pull request:
## Before this PR
check_null_field takes 4 args and only 3 are being passed when decoding union types.

chaidao001 committed:
Add type_of_union to check_null_field when decoding union types.

palantirtech commented:
Thanks for your interest in [palantir/conjure-python-client](https://github.com/palantir/conjure-python-client), @chaidao001! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

chaidao001 committed:
Add changelog manually.

I don't have the perms to use the changelog bot or create a branch on the origin repo.

chaidao001 committed:
Lint fix.
